### ISSUE TITLE
Multiline comments before functions end in same line as the function declaration

### ISSUE THREAD

**Issue by DanielMSchmidt** [prettier/prettier#4465]
**Prettier 1.12.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgBAWQJ5YCGA7nAM4QC2cAOlFltjgK4A2MAlqx1LfY1gDCVarDqZUWAGbMoYTtGkQIACgCUWYAF8QAGhAQADgqhlkoQgCdLEYgAUrCMykIA3CBwAmekACNLhGAA1nAwAMqGgTwA5sgwlsxw+jxkcJYwdgHRlITIUoSsqfoAVmQAHgBCAcGhYYTUADI8cHkFRSCRlqmWyH6Evnis0D6GljwwAOpeMAAWyAAcAAz6oxCpEwGGvaPkaa4t+pZwAI7MHEeZhNm5SPmFSSCplBxxCQ9kMaxwAIrMEPCte76GD9KaeWbIABMwICXBiwkoOV6UGgBxAzFSABV+s47qktFogA)

**Input:**
```jsx
/** My awesome
  * Multiline
  * Comment
**/ function foo() {}
```

**Output:**
```jsx
/** My awesome
 * Multiline
 * Comment
 **/ function foo() {}

```

**Expected behavior:**
```jsx
/** My awesome
 * Multiline
 * Comment
 **/ 
function foo() {}

```

**Comment by j-f1** [comment]
However, if you put a line break after the comment, it is respected:

**Prettier 1.12.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgBAWQJ5YCGA7nAM4QC2cAOlFltjgK4A2MAlqx1LfY1gDCVarDqZUdAGbMoYTtCxSIEABQBKLMAC+IADQgIABwVQyyUIQBOViMQAK1hOZSEAbhA4ATfSABGVoRgANZwMADKRkE8AObIMFbMcAY8ZHBWMPaBMZSEyFKErGkGAFZkAB4AQoEhYeGE1AAyPHD5hcUgUVZpVsj+hH54rNC+RlY8MADq3jAAFsgAHAAMBmMQaZOBRn1j5OlurQZWcACOzBzHWYQ5eUgFRckgaZQc8YmPZLGscACKzBDwNoPAwwAbTLxzZAAJhBgS4sWElFyfSg0EOIGYaQAKgMXPc0tptEA)

**Input:**
```jsx
/** My awesome
  * Multiline
  * Comment
**/
function foo() {}
```

**Output:**
```jsx
/** My awesome
 * Multiline
 * Comment
 **/
function foo() {}

```

**Comment by DanielMSchmidt** [comment]
I agree. The problem is the code above is generated by babel and I can't get a newline between the comment and the function.

**Comment by lydell** [comment]
Somewhat related: #4398
